client/components/tinymce/plugins/wplink/dialog.jsx

```diff
@@ -283,9 +304,30 @@ var LinkDialog = React.createClass( {
 						<span>{ this.translate( 'Open link in a new window/tab' ) }</span>
 					</FormLabel>
 				</FormFieldset>
+				<FormFieldset>
+					<FormLabel>
+						<span>{ this.translate( 'Link to existing content' ) }</span>
+						{ this.props.site && (
+							<PostSelector
+								siteId={ this.props.site.ID }
+								type="any"
+								status="publish"
+								orderBy="date"
+								order="DESC"
+								selected={ this.getSelectedPostId() }
+								onChange={ this.setExistingContent } />
+						) }
+					</FormLabel>
+				</FormFieldset>
 			</Dialog>
 		);
 	}
 } );
 
-module.exports = LinkDialog;
+export default connect( ( state ) => {
+	const selectedSite = getSelectedSite( state );
+	return {
+		site: selectedSite,
+		sitePosts: selectedSite ? getSitePosts( state, selectedSite.ID ) : null
+	};
+} )( LinkDialog );
```

Comment:
>Still some non es6 syntax in this file as well

Yeah, I feel dirty enough with the in-place conversions, but I tend to sneak in bits at a time :wink: I'm okay with some remaining ES5.